### PR TITLE
feat(subgraph): Add "State of Unlock" page

### PR DESCRIPTION
**Pull request by m00n620**
# Description

Subgraph updates to include proper attributes for State page

### CODE REVIEW

**Review by julien51 — COMMENTED** [review]
This looks good to me, but I want @clemsos to check it out as well since he worked on the subgraph a lot!

File: subgraph/tests/mocks.ts [hunk 1]
```diff
@@ -114,6 +115,14 @@ createMockedFunction(
   .withArgs([ethereum.Value.fromUnsignedBigInt(BigInt.fromU32(tokenId))])
   .returns([ethereum.Value.fromUnsignedBigInt(BigInt.fromU64(expiration))])
 
+createMockedFunction(
+  Address.fromString(lockAddress),
+  'createdAt',
+  'createdAt():(uint256)'
+)
```
[the remark this answers is not in the document]
Comment (m00n620) [reply subgraph/tests/mocks.ts]: resolved

File: subgraph/src/unlock.ts [hunk 2]
```diff
@@ -17,6 +17,35 @@ export function handleNewLock(event: NewLock): void {
...
+    lockStats.save()
+  }
+
+  // create lockDayData
+  let timestamp = event.block.timestamp.toI32()
+  let dayID = timestamp / 86400
+  let lockDayData = LockDayData.load(dayID.toString())
+  if (lockDayData === null) {
+    lockDayData = new LockDayData(dayID.toString())
+    lockDayData.lockDeployed = BigInt.fromI32(0)
```
[the remark this answers is not in the document]
Comment (m00n620) [reply subgraph/src/unlock.ts]: resolved

File: subgraph/abis/PublicLock.json [hunk 3]
```diff
@@ -3223,7 +3223,7 @@
       {
         "indexed": false,
         "internalType": "uint256",
-        "name": "maxKeysPerAddress",
+        "name": "maxKeysPerAcccount",
```
Comment (julien51) [inline subgraph/abis/PublicLock.json:3226]: We don't need this here, do we?

File: subgraph/schema.graphql [hunk 4]
```diff
@@ -25,4 +27,18 @@ type Key @entity {
...
+}
+
+type LockDayData @entity {
+  id: ID!
+  lockDeployed: BigInt!
+  keysSold: BigInt!
+  activeLocks: BigInt!
+}
+
+type LockStats @entity {
```
[the remark this answers is not in the document]
Comment (julien51) [reply subgraph/schema.graphql:38]: And same with `LockDayData` => `UnlockDailyData`

File: subgraph/src/public-lock.ts [hunk 5]
```diff
@@ -215,7 +230,7 @@ export function handleLockManagerRemoved(event: LockManagerRemovedEvent): void {
         newManagers.push(managerAddress)
       }
     }
-    lock.lockManagers = newManagers;
+    lock.lockManagers = newManagers
```
Comment (julien51) [inline subgraph/src/public-lock.ts:233]: Any reason for this change?

File: subgraph/schema.graphql [hunk 6]
```diff
@@ -26,3 +27,16 @@ type Key @entity {
   createdAtBlock: BigInt!
   cancelled: Boolean
 }
+
+type LockDayData @entity {
+  id: ID!
+  lockDeployed: BigInt!
+  keysSold: BigInt!
```
Comment (julien51) [inline subgraph/schema.graphql:34]: You are initializing this but not actually incrementing it

**Review by julien51 — CHANGES_REQUESTED** [review]
I thibk you are missing some calculations

File: subgraph/abis/PublicLock.json (hunk 3 above)
Comment (m00n620) [reply subgraph/abis/PublicLock.json:3226]: this is automatically changed by building smart contract.

File: subgraph/schema.graphql (hunk 6 above)
Comment (m00n620) [reply subgraph/schema.graphql:34]: fixed

File: subgraph/src/public-lock.ts (hunk 5 above)
Comment (m00n620) [reply subgraph/src/public-lock.ts:238]: just removed it to match coding style

File: subgraph/schema.graphql [hunk 7]
```diff
@@ -26,3 +27,16 @@ type Key @entity {
...
+  activeLocks: BigInt!
```
[the remark this answers is not in the document]
Comment (m00n620) [reply subgraph/schema.graphql]: we are calculating activeLocks now

File: subgraph/schema.graphql (hunk 4 above)
[the remark this answers is not in the document]
Comment (m00n620) [reply subgraph/schema.graphql:38]: resolved

File: subgraph/abis/PublicLock.json (hunk 3 above)
Comment (m00n620) [reply subgraph/abis/PublicLock.json]: resolved again

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
LGTM!

**Pull request merged**